src/ts/components/component.ts

```diff
@@ -339,7 +339,7 @@ export class Component<Config extends ComponentConfig> {
    * @param base configuration inherited from a superclass
    * @returns {Config}
    */
-  protected mergeConfig<Config>(config: Config, defaults: Config, base: Config): Config {
+  protected mergeConfig<Config>(config: Config, defaults: Partial<Config>, base: Config): Config {
```

Comment:
Not every component can fill all of its default config values. It does not need to fill out all as others are required anyway on initialization. To improve the type handling I decided to make this `Partial` here. The alternative (which was used before) was a cast to the Config type.